Add error message guidelines and how to build a PDF locally (#45)
pyansys/dev-guide: doc/source/guidelines/doc_practices.rst

across PyAnsys libraries. For more information, see :ref:`api_documentation`.

RST Files
~~~~~~~~~
To provide general usage information in your documentation, use your favorite editor to

> create RST files that you then place in the repository's ``doc\`` directory. The ``index.rst``

Reviewer: ```suggestion
create RST files that you then place in the repository's ``doc`` directory. The ``index.rst``
```

The `/` works both on Linux, Windows, and MacOS, but `\` will only work on Windows. We can also just remove this character.

Using GitBash to Clone a GitHub Repository
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> To use GitBase to clone a GitHub repository:

Reviewer: ```suggestion
To use GitBash to clone a GitHub repository:
```

   If you need to change a PR title, to its right, click the ``Edit`` button,
   which becomes a ``Save`` button while you are in editing mode.

.. _doc_building:

Building Documentation Locally
------------------------------
You can build and verify the HTML documentation for a PyAnsys library locally
by installing Sphinx and other documentation build dependencies.

Setting Up Your Machine
~~~~~~~~~~~~~~~~~~~~~~~
Anaconda provides Python and tools, such as a Python IDE (Interactive Development Environment),
a Python command line editor, and Sphinx dependencies. This gives you what you need to get up
and running.

#. Install the `Anaconda individual edition <https://www.anaconda.com/products/individual>`_.

#. If a PyAnsys library already exists, create a directory in which to place a clone of its GitHub repository.

#. Clone the PyAnsys library's GitHub repository. For more information, see :ref:`cloning`.

#. If you have not yet cloned the ``pyansys-sphinx-theme`` repository, clone it.

Installing Build Requirements
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 

> You can build documentation for the ``dev_guide`` and ``pyansys-sphinx-theme`` repositories without

Author: ```suggestion
You can build documentation for the `dev_guide <https://github.com/pyansys/dev-guide>`_ and `pyansys-sphinx-theme <https://github.com/pyansys/pyansys-sphinx-theme>`_ repositories without
```

installing a PyAnsys library in development mode. However, when you push changes that you have made
in a local branch to the GitHub repository for other PyAnsys libraries, CI checks typically require
that the full library is installed.

> #. In Anaconda Powershell, navigate to the base directory in the library's cloned repository.

Reviewer 2: ```suggestion
#. In Anaconda command prompt, navigate to the base directory in the library's cloned repository.
```

   repeating the local build process until there are no errors or obvious issues.

.. note::
   You can use ``make.bat`` to build more than HTML output. To view a summary of
   all target options, enter ``make.bat``.

Your next step is to push your changes to the library's GitHub repository
by creating a PR. For more information, see :ref:`create_pr`.

.. _pdf_building:

Building a PDF of Your Documentation Locally
--------------------------------------------
Some libraries supply a link to a PDF of the documentation under 'Assets' in the 
release notes for a version. On Linux, building this PDF can be set up as an
automated process. On Windows, building a PDF is a manual process that you run
locally:

#. Install `MiKTeX <https://miktex.org/download>`_ for Windows, selecting the
   recommended option for installing it for only your own use.
#. From the Windows Start menu, start the MiKTeX Console.
#. On the Welcome page, click ``Check for updates`` and install any available
   updates.
#. Install the latest version of Strawbery Perl, which enables you to build
   LaTeX files, accepting the default installation location (``C:\Strawberry\``).

> #. In the Command Prompt window, type ``perl-v`` to test that you installation is

Reviewer 2: ```suggestion
#. In the Command Prompt window, type ``perl-v`` to test that your installation is
```
